pytorch/imagerecognition/cifar10-resnet20-distributed-data-parallel/main.py

```diff
@@ -0,0 +1,289 @@
+"""Training ResNet for CIFAR-10 dataset.
+
+This implements the 1a image recognition benchmark task,
```

Comment:
please comment on any differences or if it implements it precisely in all aspects of the task

Reply:
It implements all of the same steps as the benchmark `cifar10-resnet20-all-reduce`. The only difference is that we wrap the model (which has the exact same definition as previously) in a DDP object and we use the SGD optimizer provided by PyTorch. DDP handles the allreduce step itself, so we don't need to use our own implementation of CentralisezSGD. More details on the implementation of DDP can be found [here](https://pytorch.org/docs/stable/notes/ddp.html). The only other change here is that we set the default behaviour to use GPU. You can see the difference in the `main.py` between the benchmark `cifar10-resnet20-all-reduce` and this one on this [link](http://www.mergely.com/k2EPXqI4/). They also use the same Dockerfile and requirements.

Reply:
perfect. want to add that info to the .py comment section?

Reply:
you mean inside the docstring here?